### ISSUE TITLE
Stashed tabs shown when option to not show stash is selected

### ISSUE THREAD

**Issue by bakj1979** [josh-berry/tab-stash#347]
### System Details

- **OS Version:** MacOS Ventura 13.0
- **Browser Version:** Firefox 110.0.1
- **Tab Stash Version:** 2.12.0.1

**Installed Browser Extensions:**

-
-

### Describe the Bug

I have selected the following behaviour in Tab Stash preferences: when the toolbar button is clicked ->> stash all (or selected tabs) and ->> don't show the stash, however when I click, the stash is indeed shown. 

### Steps to Reproduce

1. Go to about:addons
2. Click on tabstash ->> preferences
3. Click on show advanced settings"
4. Select "stash all (or selected) tabs" AND "do not show the stash"

### Expected Behavior

I expected a new tab to open (one not showing the stashed tabs).

### Additional Details

<hr>
<em>Readers: If you are also experiencing this bug, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the issue if you have more details to add, but "Me too!" comments are not counted by the voting system.</em>
<img width="1678" alt="image" src="https://user-images.githubusercontent.com/83424756/225791426-76221c79-a466-490c-a6ad-5e74bd3e0930.png">

**Comment by josh-berry** [comment]
Hi, I'm afraid I'm not able to reproduce this—with Tab Stash 2.12.0.1 in a fresh browser, I changed the options like you described, clicked the toolbar button, and as expected, the stash was not shown.

Do you happen to have Tab Stash set as your homepage or new-tab page? When Tab Stash saves all your tabs, it will replace them with a new tab if the browser window would otherwise be closed, and if Tab Stash is your homepage/new tab, that could explain what you're seeing.

If that's not the case, then it would be helpful if you could upload a screen recording, so I can see exactly what you're doing that's causing this to occur.

**Comment by bakj1979** [comment]
Attachment available until 19 Apr 2023
Hello,

Thank you for getting back to me. The issue occurs when I have restored a session (i.e. through tab stash) and then hide the tabs again. In this instance, it shows the stash.

See attached recording.

Thanks 

Jeffrey
Click to Download
 <https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAcMFPKrjKwS1iA6lLwm_H9e0QQtRAa1FYX4aIZXYpg7bIuLt858BMCCO%2F%24%7Bf%7D%3Fo%3DAiJFKrjWUUP6t054QyNPWlH9Vvtl2v2g0rXca7JmIAaw%26v%3D1%26x%3D3%26a%3DCAog8vh3AzzKhNQVzY3_zcIIFFp63YTIsIMxhEmOyRH49c8SdhDEyPzl7zAYxNj3ufkwIgEAKgkC6AMA_wcdrAFSBLRBC1FaBAEwII5qJeTwoBZ-aceH7D1bMymCx6xJRYAN25m_OnGo8oH2WRxLOCuTI3JyJTgGbXJclMAQCRHJEEufumDlh9Zeue6u4LgX8eHwzL7ZOuz6fqQ%26e%3D1681869630%26fl%3D%26r%3D711182AE-1902-4284-AB20-583EB548FCDE-1%26k%3D%24%7Buk%7D%26ckc%3Dcom.apple.largeattachment%26ckz%3D33DCC700-43D7-4826-9EA5-9D4187BDF918%26p%3D23%26s%3Dj1whOIEg3JrGtMnHcbWMUf-AmAU&uk=ez6sBwDv4tluSruGog_crQ&f=Screen%20Recording%202023-03-20%20at%2001.58.01.mov&sz=22080624>Screen Recording 2023-03-20 at 01.58.01.mov
22.1 MB

**Comment by josh-berry** [comment]
Thanks for sharing the recording, that helps a lot! I see what's happening now.

When you click the green button to restore all tabs within a group, the Tab Stash UI stays open in its own tab in the background. You can still see it on the far left in your tab bar in that recording.

Then, when you click the Tab Stash button in your toolbar, it saves all the open tabs. However, because the Tab Stash UI is still open in the background, and because Tab Stash will never stash its own tab, it *just so happens* that Tab Stash is the only remaining tab open, so that's what your browser is left showing.

In short, I suspect—and please correct me if I'm wrong—the problem here is the Tab Stash tab is *staying open* in the background when you expect it to close.

You can test this out by closing the Tab Stash tab after restoring your other tabs—in this case, clicking the toolbar button should give you a new tab (instead of the Tab Stash UI).

Hope this helps, let me know your thoughts!

**Comment by bakj1979** [comment]
Hi,

Thank you for getting back to me and spotting where I was going wrong, which in hindsight is rather obvious and intuitive!

Love the app.

Thanks again.

J

**Comment by josh-berry** [comment]
No problem at all! Glad I could help!
